### PR TITLE
fix IRSA when upgrading clusters

### PR DESCRIPTION
**Pull request by njuettner**
### What this PR does / why we need it

When clusters being upgraded they are currently broken. We need to adjust the kubeadm config so that when new nodes are joining they won't use CLOUDFRONT_PLACEHOLDER_DOMAIN anymore.

### Checklist

- [x] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/test create
/test upgrade

### CODE REVIEW

**Comment by njuettner** [comment]
Sorry I haven't seen this PR https://github.com/giantswarm/cluster-aws/pull/173, it might also work but I'd like to point out we would still miss `kubeadm-config` config map which still has the PLACEHOLDER.

File: helm/cluster-aws/files/opt/post-irsa-cloudfront.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# Replace Kubeadm Cloudfront placeholder
+CLOUDFRONT_DOMAIN="https://$(cat /etc/kubernetes/irsa/cloudfront-domain)"
+
+# We need to store cloudfront domain in kubeadmconfig config map, because kubeadm join can't inject cloudfront domain 
+kubectl --kubeconfig /etc/kubernetes/admin.conf -n kube-system get cm kubeadm-config -o yaml > /tmp/kubeadmconfig.yaml
```
Comment (calvix) [inline helm/cluster-aws/files/opt/post-irsa-cloudfront.sh:7]: AFAIK this is periodically updated by clusterAPI, so this is change will not last, or I am wrong?

**Comment by calvix** [comment]
also there seem to be some miscomunication as two people are working on same issue
https://github.com/giantswarm/cluster-aws/pull/173

File: helm/cluster-aws/files/opt/post-irsa-cloudfront.sh (hunk 1 above)
Comment (njuettner) [reply helm/cluster-aws/files/opt/post-irsa-cloudfront.sh:7]: I just ran it multiple hours, the configmap was not overwritten again 🤔

Comment (calvix) [reply helm/cluster-aws/files/opt/post-irsa-cloudfront.sh:7]: Maybe I am thinking of something else, we we are certain that the change will stays even after update than I am fine.

**Comment by njuettner** [comment]
We decided to continue using Berk's approach 👍🏻

**Pull request closed without merging**